### PR TITLE
Share component and state for InfoReview

### PR DESCRIPTION
**Pull request by adodge-bcg**
# Description

Created a share component to be displayed on information review when an applicant has an available crc to share.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested locally and with storybook.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: SPDBCSC-353

### CODE REVIEW

File: src/ecrc-frontend/src/components/page/informationreview/InformationReview.stories.js [hunk 1]
```diff
@@ -61,43 +70,102 @@ const newPayload = {
...
+                ...applicant,
+                organizationFacility: "PBS WIPB"
+              }
+            }}
+          />
+        </MemoryRouter>
+      )}
+    </FailData>
+  ))
+  .add("Display Share Mobile", () => (
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/informationreview/InformationReview.stories.js:163]: should we have a faildata version for mobile?

File: src/ecrc-frontend/src/components/page/informationreview/__snapshots__/InformationReview.test.js.snap [hunk 2]
```diff
@@ -396,6 +396,8 @@ exports[`InformationReview Component Matches the snapshot 1`] = `
           </span>
         </label>
       </section>
+      <br />
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/informationreview/__snapshots__/InformationReview.test.js.snap:399]: so in the snapshot, it doesnt seem to be rendering the share component
is this expected?

File: src/ecrc-frontend/src/components/page/informationreview/InformationReview.stories.js (hunk 1 above)
Comment (adodge-bcg) [reply src/ecrc-frontend/src/components/page/informationreview/InformationReview.stories.js:163]: FailData is wrapped around mobile. All FailData is simulating a failed check for share, so the component does not display (which is the default)

File: src/ecrc-frontend/src/components/page/informationreview/__snapshots__/InformationReview.test.js.snap (hunk 2 above)
Comment (adodge-bcg) [reply src/ecrc-frontend/src/components/page/informationreview/__snapshots__/InformationReview.test.js.snap:399]: Fixed

**Pull request merged**